### ISSUE TITLE
Wrong emulation of JSON_OBJECTAGG in Db2 and other dialects  for String values

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11398]
The current emulation for `JSON_OBJECTAGG` in Db2 is wrong:

```java
Table<Record2<String, String>> t = values(row("a", s2000), row("b", s2000), row("c", s2000)).as("t", "key", "value");
Field<String> key = t.field("key", VARCHAR);
Field<String> value = t.field("value", VARCHAR);

Record1<JSON> r =
create().select(jsonObjectAgg(key, value).returning(CLOB))
        .from(t)
        .fetchOne();
```

The resulting SQL looks like this:

```sql
select (('{' || listagg(
  ('"' || replace(
    "t"."key",
    '"',
    '\"'
  ) || '":' || cast(coalesce(
    json_value(json_object(key 'x' value "t"."value"), '$.x'),
    'null'
  ) as varchar(32672))),
  ','
) || '}'))
from (values
  ('a', 'aaaaaaaaaa...'),
  ('b', 'aaaaaaaaaa...'),
  ('c', 'aaaaaaaaaa...')
) "t" ("key", "value")
```

And the result:

```
+--------------------------------------------------+
|json_objectagg                                    |
+--------------------------------------------------+
|{"a":aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...|
+--------------------------------------------------+
```

Notice how the resulting string is not in quotes.

Affected dialects:

- [x] Db2
- [x] CockroachDB
- [x] MariaDB 10.4 and earlier
- [x] MySQL

**Comment by lukaseder** [comment]
This emulation seems to work much better, and doesn't depend on the `JSON_VALUE` hack:

```sql
select (('{' || listagg(
  regexp_replace(json_object(KEY "t"."key" value "t"."value"), '^\{(.*)\}$', '\1')
) || '}'))
from (values
  ('a', 'aaaaaaaaaa...'),
  ('b', 'aaaaaaaaaa...'),
  ('c', 'aaaaaaaaaa...')
) "t" ("key", "value")
```

**Comment by lukaseder** [comment]
This area is undergoing some change in jOOQ 3.15, which is why this fix won't be backported to 3.14
